ravenpy/utilities/data_assimilation.py

```diff
@@ -191,3 +194,173 @@ def update_state(x, qobs_pert, qobs_error, qsim):
     xa = np.maximum(xa, 0)
 
     return xa
+
+
+def perturb_full_series(model, std, start_date, end_date, dists, n_members=25):
+    """
+    This function will take a timeseries as input and perturb each variable
+    according to the desired uncertainty/confidence level and distribution of
+    each variable. Returns an Xarray Dataset object with n_members x time values
+    for each variable.
+
+    Parameters
+    ----------
+    model : Raven model class object
+        The model that will be used to perform the simulations and assimilation
+    std : tuple
+        Pairs of variable: standard deviation of the uncertainty.
+    start_date : datetime object
+        Start date of the assimilation run.
+    end_date : datetime object
+        End date of the assimilation run.
+    dists : tuple
+        Pairs of variable:distribution to identify the type of distribution each variable follows.
+    n_members : integer
+        Number of members to use in the Ensemble Kalman Filter.
+
+    Returns
+    -------
+    perturbed: path object (string)
+        Path to the perturbed hydrometeorological timeseries for the entire period.
```

Comment:
I would suggest returning the dataset, instead of writing to disk. I suggest the writing to disk should be done one level up, e.g. by the assimilation workflow.